geoh5py/data/text_data.py

```diff
@@ -39,20 +39,32 @@ def values(self) -> np.ndarray | str | None:
         """
         if (getattr(self, "_values", None) is None) and self.on_file:
             values = self.workspace.fetch_values(self)
+
+            if isinstance(values, np.ndarray) and values.dtype == object:
+                values = values.astype(str)
+
             if isinstance(values, (np.ndarray, str, type(None))):
                 self._values = values
 
         return self._values
 
     @values.setter
     def values(self, values: np.ndarray | str | None):
-        self._values = values
+        if isinstance(values, bytes):
+            values = values.decode()
 
-        if not isinstance(values, (np.ndarray, str, type(None))):
+        if isinstance(values, np.ndarray) and values.dtype == object:
+            values = values.astype(str)
+
+        if (not isinstance(values, (str, type(None), np.ndarray))) or (
+            isinstance(values, np.ndarray) and values.dtype.kind not in ["U", "S"]
```

Comment:
I had never seen this before, but makes sense. When/where are you encountering bad kinds?

Reply:
it ensure the array contains string values.
If not it raises an error. 

In the scenario someone want to push numerics into text, he should convert them before